packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -449,18 +449,23 @@ export function dispatchEventForPluginEventSystem(
               grandNode = grandNode.return;
             }
           }
-          const parentSubtreeInst = getClosestInstanceFromNode(container);
-          if (parentSubtreeInst === null) {
-            return;
-          }
-          const parentTag = parentSubtreeInst.tag;
-          // getClosestInstanceFromNode can return a HostRoot or SuspenseComponent.
-          // So we need to ensure we only set the ancestor to a HostComponent or HostText.
-          if (parentTag === HostComponent || parentTag === HostText) {
-            ancestorInst = parentSubtreeInst;
+          // Now we need to find it's corresponding host fiber in the other
+          // tree. To do this we can use getClosestInstanceFromNode, but we
+          // need to validate that the fiber is a host instance, otherwise
+          // we need to traverse up through the DOM till we find the correct
+          // node that is from the other tree.
+          while (container !== null) {
+            const parentNode = getClosestInstanceFromNode(container);
+            if (parentNode === null) {
+              return;
+            }
+            const parentTag = parentNode.tag;
+            if (parentTag === HostComponent || parentTag === HostText) {
+              node = ancestorInst = parentNode;
+              continue mainLoop;
+            }
+            container = container.parentNode;
```

Comment:
Maybe add a comment about which nodes you expect to see here. One issue is that this might be a HostRoot, but this can also be a SuspenseComponent of a dehydrated boundary. Not sure if this is correct in that case or not.

Reply:
If we hit a `SuspenseComponent`, shouldn't we also continue traversing up to find other containers as they'll still be represented correctly in the DOM, right? I tried to "break" this with dehydrated boundaries, but couldn't, so assumed the heuristcs remained consistent for this case as is for HostComponent/HostText.